pkg/volume/azure_file/azure_file.go

```diff
@@ -104,6 +105,13 @@ func (plugin *azureFilePlugin) newMounterInternal(spec *volume.Spec, pod *v1.Pod
 	if err != nil {
 		return nil, err
 	}
+	nameSpace := pod.Namespace
+	if spec.PersistentVolume != nil {
+		ns, err := getSecretNamespace(plugin.host.GetKubeClient(), spec.PersistentVolume)
```

Comment:
also, the entire block you referenced will fail on any cluster running RBAC, as nodes do not have permission to create arbitrary endpoints/services